dbms/src/Flash/Coprocessor/InterpreterUtils.cpp

```diff
@@ -162,6 +166,35 @@ void orderStreams(
     }
 }
 
+void executeLocalSort(
+    PipelineExecutorStatus & exec_status,
+    PipelineExecGroupBuilder & group_builder,
+    const SortDescription & order_descr,
+    size_t limit,
+    const Context & context,
+    const LoggerPtr & log)
+{
+    auto input_header = group_builder.getCurrentHeader();
+    if (SortHelper::isSortByConst(input_header, order_descr))
+    {
+        // For order by const and limit != 0, we will generate LimitOperator directly.
+        if (limit != 0)
+        {
+            auto local_limit = std::make_shared<LocalLimitTransformAction>(input_header, limit);
+            group_builder.transform([&](auto & builder) {
+                builder.appendTransformOp(std::make_unique<LimitTransformOp<LocalLimitPtr>>(exec_status, log->identifier(), local_limit));
+            });
+        }
+        // For order by const and limit == 0, do nothing here.
```

Comment:
Should it return an empty block if limit is zero? If so, is it expected that do nothing here?